From 0dddb4e1158705345b7f4f71635d42c89e1b0c8f Mon Sep 17 00:00:00 2001
From: Dave Anderson <anderson@redhat.com>
Date: Fri, 31 Jan 2020 17:11:46 -0500
Subject: [PATCH] Fix for an ARM64 gcc-10 compilation error.  Without the
 patch, the build of the embedded gdb module fails with an error message that
 indicates "multiple definition of 'tdesc_aarch64'". (anderson@redhat.com)

Upstream-Status: Backport [https://github.com/crash-utility/crash/commit/e770735200c02ac2414c394ea6ec5f7f033efe64]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 gdb-7.6.patch | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/gdb-7.6.patch b/gdb-7.6.patch
index f6c311d..f64b55f 100644
--- a/gdb-7.6.patch
+++ b/gdb-7.6.patch
@@ -2471,3 +2471,33 @@ diff -up gdb-7.6/opcodes/configure.orig gdb-7.6/opcodes/configure
  #else
  # error "!__i386__ && !__x86_64__"
  #endif
+--- gdb-7.6/gdb/features/aarch64.c.orig
++++ gdb-7.6/gdb/features/aarch64.c
+@@ -5,7 +5,6 @@
+ #include "osabi.h"
+ #include "target-descriptions.h"
+ 
+-struct target_desc *tdesc_aarch64;
+ static void
+ initialize_tdesc_aarch64 (void)
+ {
+--- gdb-7.6/gdb/aarch64-linux-nat.c.orig
++++ gdb-7.6/gdb/aarch64-linux-nat.c
+@@ -37,6 +37,7 @@
+ 
+ #include "gregset.h"
+ 
++extern struct target_desc *tdesc_aarch64;
+ #include "features/aarch64.c"
+ 
+ /* Defines ps_err_e, struct ps_prochandle.  */
+--- gdb-7.6/gdb/aarch64-tdep.c.orig
++++ gdb-7.6/gdb/aarch64-tdep.c
+@@ -52,6 +52,7 @@
+ #include "gdb_assert.h"
+ #include "vec.h"
+ 
++struct target_desc *tdesc_aarch64;
+ #include "features/aarch64.c"
+ #include "features/aarch64-without-fpu.c"
+ 
-- 
2.28.0

